swift/ql/src/queries/Security/ECB-Encryption/ECBEncryption.ql

```diff
@@ -0,0 +1,81 @@
+/**
+ * @name Encryption using ECB
+ * @description Highlights uses of the encryption mode 'ECB'. This mode should normally not be used because it is vulnerable to replay attacks.
+ * @kind path-problem
+ * @problem.severity warning
+ * @security-severity 7.5
+ * @precision high
+ * @id swift/ecb-encryption
+ * @tags security
+ *       external/cwe/cwe-327
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import codeql.swift.dataflow.TaintTracking
+import DataFlow::PathGraph
+
+/**
+ * An `Expr` that is used to initialize the block mode of a cipher.
+ */
+abstract class BlockMode extends Expr { }
+
+/**
+ * An `Expr` that is used to form an `AES` cipher.
+ */
+class AES extends BlockMode {
+  AES() {
+    // `blockMode` arg in `AES.init` is a sink
+    exists(ClassDecl c, AbstractFunctionDecl f, CallExpr call |
+      c.getName() = "AES" and
+      c.getAMember() = f and
+      f.getName() = ["init(key:blockMode:)", "init(key:blockMode:padding:)"] and
+      call.getStaticTarget() = f and
+      call.getArgument(1).getExpr() = this
+    )
+  }
+}
+
+/**
+ * An `Expr` that is used to form a `Blowfish` cipher.
+ */
+class Blowfish extends BlockMode {
+  Blowfish() {
+    // `blockMode` arg in `Blowfish.init` is a sink
+    exists(ClassDecl c, AbstractFunctionDecl f, CallExpr call |
+      c.getName() = "Blowfish" and
+      c.getAMember() = f and
+      f.getName() = "init(key:blockMode:padding:)" and
+      call.getStaticTarget() = f and
+      call.getArgument(1).getExpr() = this
+    )
+  }
+}
+
+/**
+ * A taint configuration from the constructor of ECB mode to expressions that use
+ * it to initialize a cipher.
+ */
+class ECBEncryptionConfig extends TaintTracking::Configuration {
```

Comment:
Should this be a `DataFlow::Configuration` instead?

Reply:
A regular `DataFlow::Configuration` might be enough here. You can mostly just do `encrypt` and `decrypt` on a `BlockMode`, which won't affect its taint status.

Reply:
Roughly speaking, `DataFlow` tracks a value from a source to a sink without alteration.  `TaintTracking` tracks any value sufficiently influenced by the source, so it will track through arithmetic operations for example.  We probably don't benefit from that here.

Reply:
Changed it to `DataFlow::Configuration` and still getting the same results. So I'm going ahead with your suggested change.